src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
+                return throwableClass.cast(t);
```

Comment:
After some experimentation on the TestNG side, I think we should have two methods:
```java
public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable);
public static <T extends Throwable> void assertThrows(Class<T> throwableClass, ThrowingRunnable runnable)
```
Here `assertThrows` delegates directly to `expectThrows` and discards its return value. I've identified three reasons why `assertThrows` should also be available:
1. It prevents warnings: discarding a Throwable return value can raise red flags
2. It facilitates convention: "assert" is called solely for its side effect, whereas "expect" returns something but first asserts something about its return value
3. It enhances discoverability: most people will look for methods prefixed with "assert", and indeed, the return value of expectThrows is often unneeded